include/lsst/afw/geom/SpanPixelIterator.h

```diff
@@ -61,7 +61,7 @@ class SpanPixelIterator : public boost::iterator_facade<SpanPixelIterator, lsst:
 
     void advance(int n) { _p.getX() += n; }
 
-    bool equal(SpanPixelIterator const& other) const {
+    bool equal(SpanPixelIterator const& other) const noexcept {
```

Comment:
If the `getX` and `getY` calls here are `noexcept`, they should be in `increment`, `decrement`, and `advance`, too, even if they're const overloads.

But also note [DM-14992](https://jira.lsstcorp.org/browse/DM-14992), which I just created after noticing the incorrect iterator tag for this class - I'm hope to do that soon, as it's a potentially serious problem, and that will probably involve this class being rewritten without `boost::iterator_facade` anyway.

Reply:
I thought that forward iterators must always be comparable (the [ForwardIterator spec](https://en.cppreference.com/w/cpp/named_req/ForwardIterator) does not mention an equality domain, while InputIterator does), while iterator moves have cases where they're allowed to throw (e.g., the spec does not define what happens if you increment a past-the-end iterator).

However, the [proposed ForwardIterator concept](https://en.cppreference.com/w/cpp/experimental/ranges/iterator/ForwardIterator) explicitly says that only comparison of iterators over the same sequence is defined. I guess that means we should remove `noexcept` here, whether or not DM-14992 is implemented.

Reply:
ForwardIterators definitely need to be equality comparable, and I don't think any kind of iterator can be comparable unless a domain is specified (see previous conversation on pointer comparisons).

In any case, my comment here was just saying that *if* `equal` can be `noexcept` *given its implementation*, then the other methods could, too, given their implementations.  I wasn't thinking about whether any possible implementation would be `noexcept`, and on those grounds what you have here is fine with me.  (Making them all `noexcept` or making none of them `noexcept` is also fine with me, as I actually don't think there is any reasonable implementation of this class that could throw when the iterator is moved, and I don't think iterator concepts require `noexcept` anywhere).